src/bsoncxx/config/prelude.hpp

```diff
@@ -16,3 +16,7 @@
 #include <bsoncxx/export.hpp>
 
 #define BSONCXX_INLINE inline __attribute__ ((__visibility__("hidden"), __always_inline__))
+
+#pragma push_macro("BSONCXX_UNREACHABLE")
+#undef BSONCXX_UNREACHABLE
+#define BSONCXX_UNREACHABLE std::abort();
```

Comment:
can you take off the semicolon here, and then put it after uses of BSONCXX_UNREACHABLE? I think it looks cleaner that way and we generally prefer to make our macros statement-like